media-source-respec.html

```diff
@@ -2719,6 +2711,280 @@ <h3>Event Summary</h3>
+            </td>
+          </tr>
+        </tbody>
+      </table>
+
+      <h3>Algorithms</h3>
+      <h4>Buffered Change</h4>
+      <p>
+        The following steps are run at the completion of all operations to the {{ManagedSourceBuffer}}
+        |buffer:ManagedSourceBuffer| that would cause a |buffer|'s {{SourceBuffer/buffered}} to change.
+        That is once {{SourceBuffer/appendBuffer()}}, {{SourceBuffer/remove()}} or
+        [=ManagedSourceBuffer/memory cleanup=] algorithm have completed.
+      </p>
+      <ol>
+        <li>
+          Let |previous buffered ranges:normalized TimeRanges| equal the
+          {{SourceBuffer/buffered}} attribute before the changes occurred.
+        </li>
+        <li>
+          Let |new buffered ranges:normalized TimeRanges| equal the new
+          {{SourceBuffer/buffered}} {{TimeRanges}}.
+        </li>
+        <li>
+          Let |added:normalized TimeRanges| equal the |previous buffered
+          ranges:normalized TimeRanges| subtracted from |new buffered
+          ranges:normalized TimeRanges|.
+        </li>
+        <li>
+          Let |removed:normalized TimeRanges| equal the |new buffered
+          ranges:normalized TimeRanges| subtracted from |previous buffered
+          ranges:normalized TimeRanges|.
+        </li>
+        <li>
+          Let |eventInitDict| be a new {{BufferedChangeEventInit}} dictionary
+          initialized with |added| as its {{BufferedChangeEventInit/addedRanges}} and
+          |removed| as its {{BufferedChangeEventInit/removedRanges}}
+        </li>
+        <li>
+          [=Queue a task=] to [=fire an event=] named {{bufferedchange}} at |buffer|
+          using the {{BufferedChangeEvent}} interface, initialized with
+          |eventInitDict|.
+        </li>
+      </ol>
+      <h4>
+        <dfn data-dfn-for="ManagedSourceBuffer">Memory cleanup</dfn>
+      </h4>
+      <ol>
+        <li>
+          <dl class="switch">
+            <dt>
+              If [=this=] is not in [=this=]'s {{ManagedMediaSource}} parent
+              {{MediaSource/activeSourceBuffers}}:
+            </dt>
+            <dd>
+              <ol>
+                <li>
+                  Run the [=coded frame removal=] algorithm with 0 and positive infinity equal to the removal range start and end timestamp respectively, and abort these steps.
+                </li>
+              </ol>
+            </dd>
+          </dl>
+        </li>
+        <li>
+          Let |removal ranges:normalized TimeRanges| equal a list of presentation
+          time ranges that can be evicted from the presentation to ensure
+          uninterrupted playback from {{HTMLMediaElement.currentTime}} until such
+          presentation could be fetched again.
+          <p class="note">
+            Implementations can use different strategies for selecting |removal ranges|
+            so web applications shouldn't depend on a specific behavior. The web
```

Comment:
I had never seen that explainer/proposal before. It's unfortunate. The aim of this blabber was to make it as vague as possible as to how it could be done and preserve the intent. 

If the aim is to make that explainer official, and have it referenced here, it needs to be made official first.

Reply:
Ok, yeah, let's consider bring the WICG proposal over as it's in scope of this work. We can probably just add it to the spec, but we would need to check what Firefox does and also what WebKit does (and how it aligns with Chrome). 

We could test for this maybe?

Reply:
Conclusion is really that authors can really depend on this... a that stuff gets evicted for (implementation specific or hardware limits) "reasons".

Reply:
To add: Firefox and Safari eviction policy are very similar:
- Evict up to the last keyframe before current time
- Evict any data in the future not contiguous in the buffer ahead of currentTime

Reply:
Filed bug for followup #341 ... treating this as resolved for now.